### PR TITLE
Release/wanderer

### PR DESCRIPTION
**Pull request by jtbaird**
# PR Details
## Description
This PR is to merge the Wanderer release (carma-system-3.4.0) into the master branch.

## Related Issue
None

## Motivation and Context
Purpose of the Wanderer release is to incorporate four major enhancements into the CARMA system. First, a steering limiter was added in order to prevent Carma from taking turns that present a risk of vehicle rollover. Second, updates were made to the health_monitor to allow for graceful handling of sensor failure, including control handover to the human driver. Third, carma-messenger was updated to detect nearby automated vehicles via DSRC. And fourth, functionality was added to carma-messenger to allow for requesting, receiving, and displaying truck safety info from a passing Carma truck via DSRC while vehicles are in motion. Along with the above enhancements, several bug fixes and CI related enhancements are included in this release.

## How Has This Been Tested?
Per the release test plan.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -52,7 +52,7 @@ jobs:
           name: Pull Deps
           command: |
             source ${INIT_ENV}
-            ./src/CARMAPlatform/docker/checkout.sh ${PWD}
+            ./src/CARMAPlatform/docker/checkout.bash -r ${PWD}
```
Comment (TonysCousin) [inline .circleci/config.yml:55]: I believe this directory should be carma-platform.

File: .circleci/config.yml [hunk 2]
```diff
@@ -67,22 +67,18 @@ jobs:
...
-            ./gradlew testReport --info
       - run:
           name: Save Java Test results
           command: |
             source ${INIT_ENV}
-            mkdir -p ~/junit/ 
-            sudo mkdir -p /reports/tests
-            sudo mv src/CARMAPlatform/carmajava/build/reports/allTests/* /reports/tests/
-            find . -type f -regex ".*/build/test-results/.*xml" -exec cp {} ~/junit/ \;
+            if [ -z "$(ls -A src/CARMAPlatform/carmajava/build/reports/allTests)" ]; then
```
Comment (TonysCousin) [inline .circleci/config.yml:74]: Same here and on line 79 below?  With all these in place, maybe I'm wrong?

File: .circleci/config.yml (hunk 1 above)
Comment (jtbaird) [reply .circleci/config.yml:55]: So when we were doing the name change we discussed this and decided there wasn't a need to change the directory names used in CI, but we can change it now if we'd like. The dir is right though and is created two steps prior in config.yml

Comment (TonysCousin) [reply .circleci/config.yml:55]: No need.  Just wanted to make sure it's not broken.

**Review by TonysCousin — APPROVED** [review]

**Pull request merged**